### PR TITLE
[LM-11317] Added local storage with hive

### PR DESCRIPTION
**Pull request by AnujLM**
## Description

**Features**
1. Added local storage with hive

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: lib/packages/core/lib/src/core/core.dart [hunk 1]
```diff
@@ -44,9 +45,14 @@ class LMChatCore {
     if (domain != null) _clientDomain = domain;
     chatConfig = config ?? LMChatConfig();
     if (widgets != null) _widgetUtility = widgets;
-    await LMChatPreferences.instance.initialize();
     LMChatTheme.instance.initialise(theme: theme);
+    LMResponse initDB = await this.lmChatClient.initiateDB();
```
Comment (ishaan1607) [inline lib/packages/core/lib/src/core/core.dart:49]:
```suggestion
    LMResponse isDBInitiated = await this.lmChatClient.initiateDB();
```

File: lib/packages/core/lib/src/utils/preferences/preferences.dart [hunk 2]
```diff
@@ -23,63 +20,35 @@ class LMChatPreferences extends ILMPreferenceService {
...
-  User? currentUser;
-
-  set setCurrentUser(User user) => currentUser = user;
-  get getCurrentUser => currentUser;
-
-  @override
-  Future<void> initialize() async {
-    _sharedPreferences = await SharedPreferences.getInstance();
-  }
+  get getCurrentUser => getUser();
```
Comment (ishaan1607) [inline lib/packages/core/lib/src/utils/preferences/preferences.dart:25]: add comments

File: lib/packages/core/lib/src/utils/preferences/preferences.dart [hunk 3]
```diff
@@ -23,63 +20,35 @@ class LMChatPreferences extends ILMPreferenceService {
...
 
   @override
   Future<void> storeUserData(User user) async {
-    setCurrentUser = user;
-    UserEntity userEntity = user.toEntity();
-    Map<String, dynamic> userData = userEntity.toJson();
-    String userString = jsonEncode(userData);
-    await _sharedPreferences.setString('user', userString);
+    await LMChatCore.instance.lmChatClient.deleteLoggedInUser();
+    await LMChatCore.instance.lmChatClient.insertOrUpdateLoggedInUser(user);
```
Comment (ishaan1607) [inline lib/packages/core/lib/src/utils/preferences/preferences.dart:30]: please don't delete the user, it is irrelevant in real world scenario

File: lib/packages/core/lib/src/views/home/home_feed.dart [hunk 4]
```diff
@@ -129,6 +130,26 @@ class _LMChatHomeScreenState extends State<LMChatHomeScreen> {
     return AnnotatedRegion<SystemUiOverlayStyle>(
       value: SystemUiOverlayStyle.dark,
       child: Scaffold(
+        floatingActionButton: FloatingActionButton(
+          onPressed: () async {
+            debugPrint("Floating action button pressed");
```
Comment (ishaan1607) [inline lib/packages/core/lib/src/views/home/home_feed.dart:135]: remove debugPrint()

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (divyanshgandhilm) [inline .vscode/settings.json:1]: Let's omit this file

File: lib/packages/core/lib/src/core/core.dart [hunk 6]
```diff
@@ -5,6 +5,7 @@ import 'package:likeminds_chat_flutter_core/src/utils/notifications/notification
 import 'package:likeminds_chat_flutter_core/src/utils/preferences/preferences.dart';
 import 'package:likeminds_chat_flutter_core/src/utils/utils.dart';
 import 'package:likeminds_chat_flutter_ui/likeminds_chat_flutter_ui.dart';
+import 'package:flutter/foundation.dart';
```
Comment (divyanshgandhilm) [inline lib/packages/core/lib/src/core/core.dart:8]: I don't see any use of this import

File: lib/packages/core/lib/src/core/core.dart [hunk 7]
```diff
@@ -44,9 +45,14 @@ class LMChatCore {
     if (domain != null) _clientDomain = domain;
     chatConfig = config ?? LMChatConfig();
     if (widgets != null) _widgetUtility = widgets;
-    await LMChatPreferences.instance.initialize();
     LMChatTheme.instance.initialise(theme: theme);
+    LMResponse initDB = await this.lmChatClient.initiateDB();
+    if (!initDB.success) {
+      return LMResponse.error(
+          errorMessage: initDB.errorMessage ?? "Error in initiating DB");
```
Comment (divyanshgandhilm) [inline lib/packages/core/lib/src/core/core.dart:52]: "Error in setting up local storage"

File: lib/packages/core/lib/src/views/home/home_feed.dart (hunk 4 above)
Comment (divyanshgandhilm) [reply lib/packages/core/lib/src/views/home/home_feed.dart:135]: Was this code just for testing? As I see all of this removed in the next PR anyway.

File: lib/packages/core/lib/src/core/core.dart (hunk 1 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/core/core.dart:49]: Done

File: lib/packages/core/lib/src/utils/preferences/preferences.dart (hunk 2 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/utils/preferences/preferences.dart:25]: done

File: lib/packages/core/lib/src/utils/preferences/preferences.dart (hunk 3 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/utils/preferences/preferences.dart:30]: Removed.

File: lib/packages/core/lib/src/views/home/home_feed.dart (hunk 4 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/views/home/home_feed.dart:135]: yes it was for testing and removed in next PR

File: .vscode/settings.json (hunk 5 above)
Comment (AnujLM) [reply .vscode/settings.json:1]: done

File: lib/packages/core/lib/src/core/core.dart (hunk 6 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/core/core.dart:8]: removed

File: lib/packages/core/lib/src/core/core.dart (hunk 7 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/core/core.dart:52]: done

**Comment by AnujLM** [comment]
Merging this PR, changes are done in #15

**Pull request merged**